Replaced ActionableMessage Instances in edit-gas

pritam1813 opened the pull request:
## Explanation
- This PR is part of #19528 
- This PR replaces instances of `ActionableMessage` with `BannerAlert` in `ui/components/app/edit-gas-display/edit-gas-display.component.js`

## Screenshots/Screencaps

<!-- If you're making a change to the UI, make sure to capture a screenshot or a short video showing off your work! -->

### Before

<!-- How did the UI you changed look before your changes? Drag your file(s) below this line: -->
![Before](https://github.com/MetaMask/metamask-extension/assets/46365255/f8753a8e-2142-4764-9319-e863c6913b62)
![Before2](https://github.com/MetaMask/metamask-extension/assets/46365255/1632085b-8de9-4f9e-a752-51d4d2b92d37)


### After
<img width="448" alt="Screenshot 2023-07-28 at 4 11 20 PM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/8334d551-490d-4123-9033-8ccad8339741">
<img width="488" alt="Screenshot 2023-07-28 at 4 13 26 PM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/ef902869-f323-4ca7-8277-51b3075818d6">

## Manual Testing Steps

1. Build the project with `yarn dist`
2. Install the build to browser
3. Go to [Metamask Dapp Test Page](https://metamask.github.io/test-dapp/) and connect the wallet.
4. Click **SEND LEGACY TRANSACTION**
5. Edit the **Estimated gas fee**
6. Right click and Inspect the warning element.

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

pritam1813 committed:
Replaced ActionableMessage Instances in edit-gas

pritam1813 commented:
Hi @georgewrmarshall , please review this.

georgewrmarshall reviewed (changes requested):
Hey @pritam1813, this is looking good but I believe the after screenshots in the before/after section is not a good representation of the changes. It looks like the same screenshots. Could you please update the After section with the correct screenshots and also add some manual testing steps. Thanks

pritam1813 commented:
Hi @georgewrmarshall , Sorry , I thought that there isn't much visual difference in `BannerAlert` with `Severity` warning and `ActionableMessage` with type `Danger`, but,  actually it was due to the `className="actionable-message--warning"` which was interfering with the `BannerAlert` styling. I have changed the `className` to `"banner-alert--warning"`, I hope that's OK. 

 I have also changed the `ErrorMessage` element to `BannerAlert` `Severity='danger'`, which was displaying the insufficient balance error message.

pritam1813 committed:
Updated BannerAlert className

pritam1813 committed:
Changed errorKey to errorMessage

georgewrmarshall committed:
Removing CSS and restoring local key

georgewrmarshall commented on ui/components/app/edit-gas-display/edit-gas-display.component.js:
```diff
@@ -62,38 +61,30 @@ export default function EditGasDisplay({
 
   const showTopError = balanceError && txParamsHaveBeenCustomized;
 
-  let errorKey;
-  if (balanceError) {
-    errorKey = 'insufficientFunds';
-  }
```
Moved to line `70`

georgewrmarshall commented on ui/components/app/edit-gas-display/edit-gas-display.component.js:
```diff
@@ -62,38 +61,30 @@ export default function EditGasDisplay({
     <div className="edit-gas-display">
       <div className="edit-gas-display__content">
         {showTopError && (
-          <div className="edit-gas-display__warning">
```
Can reduce nesting and utilize style utility props for margin styles

georgewrmarshall commented on ui/components/app/edit-gas-display/index.scss:
```diff
@@ -1,13 +1,4 @@
 .edit-gas-display {
-  & .actionable-message--warning,
-  & .actionable-message--error {
-    margin-top: 0;
-
-    & .actionable-message__message {
-      text-align: start;
-    }
-  }
```
No longer using the components that use these class names

georgewrmarshall commented on ui/components/app/edit-gas-display/index.scss:
```diff
@@ -21,10 +12,6 @@
     }
   }
 
-  &__dapp-acknowledgement-warning {
-    margin-bottom: 20px;
-  }
```
No longer using this classname
![Screenshot 2023-07-28 at 4 08 41 PM](https://github.com/MetaMask/metamask-extension/assets/8112138/bfd2f375-3bb2-4d91-858d-7ac93ffe7902)

georgewrmarshall commented on ui/components/app/edit-gas-display/index.scss:
```diff
@@ -39,35 +26,10 @@
     margin: 20px auto;
   }
 
-  &__advanced-button {
```
No classname exists for these styles
![Screenshot 2023-07-28 at 3 54 46 PM](https://github.com/MetaMask/metamask-extension/assets/8112138/86eeb862-d216-4590-91cc-61c709d1aabe)

georgewrmarshall commented on ui/components/app/edit-gas-display/index.scss:
```diff
@@ -39,35 +26,10 @@
     margin-top: 20px;
   }
 
-  &__education {
```
No className exists for these styles
![Screenshot 2023-07-28 at 3 54 21 PM](https://github.com/MetaMask/metamask-extension/assets/8112138/fe9cfd8e-5748-4e31-beae-23b60b69f4c8)

georgewrmarshall commented on ui/components/app/edit-gas-display/index.scss:
```diff
@@ -39,35 +26,10 @@
-    }
-  }
-
-  &__warning {
```
Can be replaced with style utility props

georgewrmarshall reviewed (approved):
LGTM! While I was in your branch I removed some of the unused styles. I've updated the PR description with up to date screenshots. Thanks for you contribution @pritam1813

georgewrmarshall committed:
Merge branch 'develop' into part-of-#19528-in-edit-gas-display
